### DISCUSSION TITLE
Does calling `sentry_sdk.init` with empty dsn string initialize sentry?

### DISCUSSION THREAD

**Discussion by randspace0** [getsentry/sentry-python#1614] [Q&A] [answered]
The title is the question

```
sentry_sdk.init(
    dsn="",
    integrations=[DjangoIntegration()]
)
```

**Comment by sl0thentr0py** [comment, marked as answer]
hey @andraantariksa 

The sdk is technically initialized but [no transport is made](https://github.com/getsentry/sentry-python/blob/0e6aa6d83b3cebdaec98c98d2e873cba41d9893a/sentry_sdk/transport.py#L528-L529) so there will be no sending of data to sentry.

Everything basically no-ops.

**Reply by randspace0** [reply]
So it still has a runtime overhead, right? Especially when I'm testing my Django app (related with https://github.com/getsentry/sentry-python/issues/668)

**Reply by sl0thentr0py** [reply]
well there will be _some_ overhead sure, but not as much as using all of Sentry turned on. Most things should no-op, but there might be some one-off code paths that run/create useless objects. We haven't optimized everything fully yet on that front.

**Reply by randspace0** [reply]
Thanks!
